src/flake8_boolean_trap/plugins.rs

```diff
@@ -0,0 +1,75 @@
+use rustpython_ast::{Arguments, ExprKind};
+use rustpython_parser::ast::{Constant, Expr, Located};
+
+use crate::ast::types::Range;
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckKind};
+
+fn is_boolean_arg(arg: &Located<ExprKind>) -> bool {
+    matches!(
+        &arg.node,
+        ExprKind::Constant {
+            value: Constant::Bool(_),
+            ..
+        }
+    )
+}
+
+fn add_if_boolean(checker: &mut Checker, arg: &Located<ExprKind>, kind: CheckKind) {
+    if is_boolean_arg(arg) {
+        checker.add_check(Check::new(kind, Range::from_located(arg)));
+    }
+}
+
+pub fn check_positional_boolean_in_def(checker: &mut Checker, arguments: &Arguments) {
+    for arg in arguments.args.iter().chain(arguments.posonlyargs.iter()) {
+        let hint;
+        if arg.node.annotation.is_none() {
+            continue;
+        }
+
+        if let Some(expr) = &arg.node.annotation {
+            match &expr.node {
+                ExprKind::Name { id, .. } => {
+                    hint = id.to_string();
+                }
+                ExprKind::Constant { value, .. } => {
+                    hint = value.to_string();
+                }
+                _ => {
+                    continue;
+                }
+            };
+            // check for both bool (python class) and 'bool' (string annotation)
+            if ["bool", "'bool'"].iter().any(|v| v == &hint) {
+                checker.add_check(Check::new(
+                    CheckKind::BooleanPositionalArgInFunctionDefinition,
+                    Range::from_located(arg),
+                ));
+            }
+        }
+    }
+}
+
+pub fn check_boolean_default_value_in_function_definition(
+    checker: &mut Checker,
+    arguments: &Arguments,
+) {
+    for arg in arguments.defaults.iter() {
```

Comment:
Do you need to check `kw_defaults` here too?

Reply:
kw_defaults are fine as booleans (in fact that would be a valid ast fix in an upcoming release), as they do not allow positional calls. MWE:

```py
def fun(a, *, b=True):
  ...
fun(4, True) # raises TypeError
```